src/plugins/mdraid.c

```diff
@@ -193,8 +194,16 @@ static GHashTable* parse_mdadm_vars (const gchar *str, const gchar *item_sep, co
         if (g_strv_length ((gchar **) key_val) == 2) {
             /* we only want to process valid lines (with the separator) */
             /* only use the first value for the given key */
-            if (!g_hash_table_contains (table, g_strstrip (key_val[0])))
-                g_hash_table_insert (table, g_strstrip (key_val[0]), g_strstrip (key_val[1]));
+            if (!g_hash_table_contains (table, g_strstrip (key_val[0]))) {
+                vals = g_strsplit (key_val[1], "<--", 2);
+                if (g_strv_length ((gchar **) vals) == 2)
```

Comment:
I think the ``if`` should first check if the ``"<--"`` string is in ``key_val[1]``, decide on that and then do the ``g_strsplit()`` (as well as ``g_free()`` the second value) only in the particular branch.